### PR TITLE
Bump minimum GDAL version to 2.0, remove old version #ifdefs

### PR DESCRIPTION
**Pull request by nyalldawson**
Based on https://github.com/qgis/QGIS/commit/6c5ec70f8782808fc5b35fc87ab1dac3825aa0e9#commitcomment-20335255 it seems there's consensus to bump the minimum GDAL version to 2.0 and drop distros which do not provide this (Ubuntu 16.04)

This PR implements this bump.

### CODE REVIEW

**Comment by rouault** [comment]
LGTM. Thanks @nyalldawson for the cleanup ! 
Note: the INSTALL file at root still mentions GDAL >= 1.4

File: src/analysis/raster/qgskde.cpp [hunk 1]
```diff
@@ -23,11 +23,7 @@
 #define M_PI 3.14159265358979323846
 #endif
 
-#if defined(GDAL_VERSION_NUM) && GDAL_VERSION_NUM >= 1800
 #define TO8F(x) (x).toUtf8().constData()
```
Comment (jef-n) [inline src/analysis/raster/qgskde.cpp]: resolve this too?

File: src/app/qgisapp.cpp [hunk 2]
```diff
@@ -266,9 +266,7 @@
 #include <gdal_version.h>
 #include <proj_api.h>
 
-#if defined(GDAL_COMPUTE_VERSION) && GDAL_VERSION_NUM >= GDAL_COMPUTE_VERSION(1,11,0)
 #define SUPPORT_GEOPACKAGE
```
Comment (jef-n) [inline src/app/qgisapp.cpp]: another one?

File: src/gui/qgsnewgeopackagelayerdialog.cpp [hunk 3]
```diff
@@ -42,14 +42,12 @@
 #include <cpl_error.h>
 #include <cpl_string.h>
 
-#if defined(GDAL_COMPUTE_VERSION) && GDAL_VERSION_NUM >= GDAL_COMPUTE_VERSION(2,0,0)
 #define SUPPORT_GEOMETRY_LESS
```
Comment (jef-n) [inline src/gui/qgsnewgeopackagelayerdialog.cpp]: and more...

File: src/analysis/raster/qgskde.cpp (hunk 1 above)
Comment (nyalldawson) [inline src/analysis/raster/qgskde.cpp]: Fixed

File: src/app/qgisapp.cpp (hunk 2 above)
Comment (nyalldawson) [inline src/app/qgisapp.cpp]: Fixed

File: src/gui/qgsnewgeopackagelayerdialog.cpp (hunk 3 above)
Comment (nyalldawson) [inline src/gui/qgsnewgeopackagelayerdialog.cpp]: Fixed

**Pull request merged**